plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -140,6 +142,7 @@
     static final int MEMSQL_VARCHAR_MAX_LENGTH = 21844;
     static final int MEMSQL_TEXT_MAX_LENGTH = 65535;
     static final int MEMSQL_MEDIUMTEXT_MAX_LENGTH = 16777215;
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```

Comment:
Use `uuuu`. IIRC `yyyy` is year-of-era so it will give incorrect results for dates before 0 A.D.

Or add an optional `G` to include the era.

Reply:
Good catch. I thought `yyyy` is fine because MemSQL doesn't support dates before 0 A.D., but it will lead to incorrect predicates as #10208. Unfortunately, we can't capture this using SqlDataTypeTest.